pages/signup/index.tsx

```diff
@@ -1,86 +1,129 @@
-import React from 'react'
+//import libraries
+import React, { useState } from 'react'
 import { Formik, Form, Field } from 'formik'
+import Router from 'next/router'
 
+//import components
 import Card from '../../components/Card'
 import Layout from '../../components/Layout'
 import Input from '../../components/Input'
+
+//import helpers
 import { signupValidation } from '../../helpers/formValidation'
-import { Props } from '../../@types/signup'
+import { signupUser } from '../../helpers/signupUser'
+
+//import types
+import { Props, Values, SignupErrors } from '../../@types/signup'
 
-const initialValues = {
+const initialValues: Values = {
   email: '',
   username: '',
   password: '',
   firstName: '',
   lastName: ''
 }
 
-const Signup: React.FC<Props> = props => (
-  <Layout>
-    <Card title="Create Account">
-      <Formik
-        validateOnBlur
-        initialValues={initialValues}
-        validationSchema={signupValidation}
-        onSubmit={values => props.submitSignup(values)}
-      >
-        <Form data-testid="form">
-          <div className="form-group ">
-            <Field
-              name="email"
-              placeholder="Email address"
-              type="email"
-              data-testid="email"
-              as={Input}
-            />
+const ErrorMessage = (signupErrors: SignupErrors) => {
+  return (
+    <div className="bg-light m-auto px-5 border-0'">
+      {signupErrors.signupErrors.confirmEmail ? (
+        <h5 className="text-danger">
+          An account has already been registered with this email
+        </h5>
+      ) : (
+        ''
+      )}
+      {signupErrors.signupErrors.userName ? (
+        <h5 className="text-danger">Username is already taken</h5>
+      ) : (
+        ''
+      )}
+    </div>
+  )
+}
 
-            <Field
-              name="username"
-              placeholder="Username"
-              data-testid="username"
-              as={Input}
-            />
+const Signup: React.FC<Props> = () => {
+  const [signupErrors, setSignupErrors] = useState({})
+  const handleSubmit = async (values: Values) => {
+    const data = await signupUser(values)
+    if (data.success) {
+      Router.push('/signup/success')
+    } else {
+      setSignupErrors(Object.assign({}, data.errorMessage))
+    }
+  }
+  return (
+    <Layout>
+      <Card title="Create Account">
+        {Object.keys(signupErrors).length > 0 ? (
+          <ErrorMessage signupErrors={{ ...signupErrors }} />
+        ) : (
+          <></>
+        )}
```

Comment:
this logic could be handled directly by the `ErrorMessage` component to keep the main component clean.

```
<Card title="Create Account">
  <ErrorMessage signupErrors={{...signupErrors}} />
  ....
```